### DISCUSSION TITLE
How to get the real client IP from the container inside WSL2?

### DISCUSSION THREAD

**Discussion by billychow** [microsoft/WSL#7087] [Q&A]
Hello everyone,

I have an instance of gitlab running on WSL2 and Docker.
Everything is working fine, except that it always identifies the client IP as 127.0.0.1, even if it is accessed from the internet.
So is there any way to fix it? 

Thanks in advance!

**Comment by therealkenc** [comment]
There is no indication what is meant by "the client" here, but in general on WSL you can get the WSL (ie Linux) side with `hostname -I` and the Windows side with `ip addr show dev eth0`. Or by [other means](https://docs.microsoft.com/en-us/windows/wsl/compare-versions#accessing-network-applications).

**Comment by tmknight** [comment]
Reviving as I'm seeing the same issue.  To clarify:
Server=docker container running host application residing in WSL2 (e.g. nginx)
Client=external 'guest' calling host application (e.g. the machine calling nginx from the internet)

In my scenario, the IP address of the 'client' is ALWAYS seen as ::ffff:172.18.0.1, which is a private address space/non-routable subnet used by WSL2.  Thus the 'real' IP address (e.g. 92.0.0.1) of the machine making the connection is not seen/known by the host application.

How do we ensure the 'real' IP address is passed to the host application?

Similar: https://github.com/docker/for-win/issues/12644

**Comment by LIFsCode** [comment]
Hello, I have the same issue. Is there already any solution or workaround for this?
